multihomes.dsc

```diff
@@ -0,0 +1,214 @@
+        - if <player.flag[mutehomes.sethome]||false>:
+          - narrate "the messages that used to appear when doing /sethome will appear again" format:zc_home_text
+          - flag player mutehomes.sethome:false
+        - else:
+          - narrate "The messages that appear when doing /sethome will no longer appear" format:zc_home_text
+          - flag player mutehomes.sethome:true
+      - default:
+        - narrate "Put either home, delhome or sethome after <&a><&click[/mutehome<&sp>].type[suggest_command]>/mutehome<&end_click>" format:zc_home_text
+#-----------------------------------#
+#add (playername):(homename) support :check:
+#-----------------------------------#
+zc_home_text:
+    type: format
+    debug: false
+    format: <&color[#009900]>Z<&color[#00CC00]>c <&color[#00FF00]>H<&color[#33FF33]>o<&color[#66FF66]>m<&color[#99FF99]>e<&color[#CCFFCC]>s <&7><&gt><&gt><&f> <text>
+
+home_command:
+  type: command
+  name: home
+  description: used to tp to a home you created with /sethome
+  usage: /home (name)
+  debug: false
+  tab completions:
+    1: <player.flag[homes].keys>
+  script:
+    - if <context.args.first.contains_any_text[:]||false>:
+      - if !<player.has_permission[zc.homes_admin]>:
+        - narrate "You cannot perform this command." format:zc_home_text
+        - stop
+      - define player <server.match_offline_player[<context.args.first.before[:]>]||noone>
+      - define name <context.args.first.after[:]||default>
+      - define is_admin true
+      - if <[player]> == noone:
+        - narrate "<context.args.first.before[:]> Does not seem to be a player" format:zc_home_text
+        - stop
+    - else:
+      - define player <player>
+      - define name <context.args.first||default>
+      - define is_admin false
+    - if !<[player].flag[homes].keys.contains[<[name]>]||false>:
+      - narrate "You cannot teleport to a home <tern[<[is_admin]>].pass[<[player].name> does].fail[you do]> not have" format:zc_home_text
+      - stop
+    - if !<server.worlds.contains[<[player].flag[homes.<[name]>].world>]>:
+      - narrate "The world this home was in has been unloaded" format:zc_home_text
+      - stop
+    - teleport <player> <[player].flag[homes.<[name]>]>
+    - if <player.flag[mutehomes.home]||false>:
+      - stop
+    - narrate "you've been teleported to <tern[<[is_admin]>].pass[<[player].name>'s <[name]>].fail[<[name]>]>" format:zc_home_text
+
+#----------------------------------#
+#make sure trusted+ get unlimited homes and normies just one :check:
+#----------------------------------#
+sethome_command:
+  type: command
+  name: sethome
+  description: used to set your home at a location you are currently at
+  usage: /sethome (name)
+  debug: false
+  script:
+    - if <context.args.is_empty>:
+      - define name default
+    - else:
+      - define name <context.args.first.strip_color>
+    - if !<[name].to_lowercase.matches_character_set[abcdefghijklmnopqrstuvwxyz1234567890_/-]>:
```

Comment:
allowing symbols but not capitals feels weird, just a suggestion that i feel makes sense
```suggestion
    - if !<[name].to_lowercase.matches_character_set[abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890_/-]>:
```

Reply:
resolved in discord